test/functional/lua/vim_spec.lua

```diff
@@ -775,6 +775,63 @@ describe('lua stdlib', function()
     eq(NIL, funcs.luaeval "vim.g.nonexistant")
   end)
 
+  it('vim.b', function()
+    exec_lua [[
+    vim.api.nvim_buf_set_var(0, "testing", "hi")
+    vim.api.nvim_buf_set_var(0, "other", 123)
+    ]]
+
+    eq('hi', funcs.luaeval "vim.b.testing")
+    eq(123, funcs.luaeval "vim.b.other")
+    eq(NIL, funcs.luaeval "vim.b.nonexistant")
+
+    exec_lua [[
+    vim.cmd "vnew"
+    ]]
+
+    eq(NIL, funcs.luaeval "vim.b.testing")
```

Comment:
Should the tests also include setting/deleting variables?

Reply:
Ah, yeah. Why not. I can add it.